hostinfo/hostinfo_linux_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux && !android
+// +build linux,!android
+
+package hostinfo
+
+import (
+	"testing"
+)
+
+func TestQnap(t *testing.T) {
+	version_info := `commit 2910d3a594b068024ed01a64a0fe4168cb001a12
+Date:   2022-05-30 16:08:45 +0800
+================================================
+* QTSFW_5.0.0
+remotes/origin/QTSFW_5.0.0`
+
+	got := getQnapQtsVersion(version_info)
+	want := "QTS 5.0.0"
+	if got != want {
+		t.Errorf("got %v; want %v", got, want)
```

Comment:
use %q for failing strings. %q shows empty, trailing whitespace, binary escaped, etc.

(here and elsewhere)

Reply:
Done.